### DISCUSSION TITLE
Deploy/Validate Commands Should Provide a URL back to the Salesforce Deployment Status portal

### DISCUSSION THREAD

**Discussion by mcarvin8** [forcedotcom/cli#2531] [Ideas]
When a deployment/validation is started by the Salesforce CLI, the Deploy ID is logged.

To track the deployment status in Salesforce directly, you have to take that ID and then find it in the org manually.

It would be nice for the CLI to automatically print the URL back to the Salesforce Deploy Status for that specific deploy ID in the org. 

I have a custom Python script that creates & prints the Classic URL path, but I thought this would be a nice feature for the CLI to handle so users have an easy re-direct back to Salesforce in case they want to monitor the deployment there or the CLI has a failure, requiring users to go to Salesforce to confirm if the deploy passed in the org.

The below screenshot shows how I accomplished this with my Python script:
![image](https://github.com/forcedotcom/cli/assets/90224411/167ce62f-c8a0-48ba-9814-acbc92a16103)

**Comment by cristiand391** [comment]
I thought about this when playing with TUIs last year, but instead of printing the URL it would allow a few shorcuts like `Press CTRL-o to open the deploy in the browser`, might try again with [ink](https://github.com/vadimdemedes/ink) as a POC.

Another option could be to make the deploy ID a clickable link using something like:
https://github.com/sindresorhus/terminal-link

but that only covers users actually running the command, it would a way to easily open them from logs.
**EDIT**:
maybe print it if env var`CI=true`? 🤷🏼

**Reply by mcarvin8** [reply]
Yea not a bad idea using a new environment variable for CI environments. That way, any user can click it from the pipeline.

**Comment by mcarvin8** [comment]
Closing as complete. My PR has been accepted into the Plugin Deploy Retrieve repo to add this feature.
